taichi/backends/opengl/shaders/print.glsl.h

```diff
@@ -0,0 +1,32 @@
+// vim: ft=glsl
+// clang-format off
+#include "taichi/util/macros.h"
+STR(
+int _msg_allocate_slot() {
+  return atomicAdd(_msg_count_, 1);
+}
+
+void _msg_set(int msgid, int contid, int type, int value) {
+  int base = msgid << 5; // MSG_SIZE = (1 << 5)
+  _mesg_i32_[base + contid] = value;
+  _mesg_i32_[base + 30] |= (type & 1) << contid;
+  _mesg_i32_[base + 31] |= (type >> 1) << contid;
```

Comment:
`base + 30` for `type_bitmap_low`.
`base + 31` for `type_bitmap_high`.
combine together, obtains `type`.
I know it looks trick here, but it enable us to print 29 contents, instead of just 16 contents. And yet I don't find a better solution to utilize memory.

Reply:
Update: I improved readability in C++ version of `GLSLRuntime`, so although limitations make me can't improve GLSL version, reviewers still can get my point in the C++ version.